src/screens/DiscoveryScreenV2/fragment/DomainFragment.js

```diff
@@ -262,20 +308,33 @@ const DomainFragment = ({
     });
     const data = isFirstTimeOpen
       ? [
-          ...followingDomains.map((item, index) => ({
-            ...item,
-            user_id_follower: item.user_id_follower
-          })),
           {separator: true},
-          ...unfollowingDomains.map((item, index) => ({
+          ...unfollowingDomains.map((item) => ({
             ...item,
             user_id_follower: item.user_id_follower
           }))
         ]
-      : [...newMapFollowedDomain, {separator: true}, ...newMapUnfollowedDomain];
+      : [{separator: true}, ...newMapUnfollowedDomain];
+
+    const firstData = isFirstTimeOpen
+      ? followingDomains.map((item) => ({
+          ...item,
+          user_id_follower: item.user_id_follower
+        }))
+      : newMapFollowedDomain;
 
     return (
       <FlatList
+        ListHeaderComponent={() => (
+          <>
+            <RecentSearch
+              shown={!withoutRecent || isFirstTimeOpen}
+              setSearchText={setSearchText}
+              setIsFirstTimeOpen={setIsFirstTimeOpen}
+            />
+            <AccordionView data={firstData} />
+          </>
+        )}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The data manipulation logic for `firstData` and `data` seems a bit complex and hard to understand. It would be better to extract this logic into separate functions for better readability and maintainability. Also, consider adding comments to explain what's happening in these operations.

```diff
- const data = isFirstTimeOpen
-   ? [
-        {separator: true},
-       ...unfollowingDomains.map((item) => ({
-          ...item,
-          user_id_follower: item.user_id_follower
-        }))
-      ]
-   : [{separator: true}, ...newMapUnfollowedDomain];
-
- const firstData = isFirstTimeOpen
-   ? followingDomains.map((item) => ({
-       ...item,
-       user_id_follower: item.user_id_follower
-     }))
-   : newMapFollowedDomain;

+ const getData = () => {
+   if (isFirstTimeOpen) {
+     return [
+       {separator: true},
+       ...unfollowingDomains.map((item) => ({
+         ...item,
+         user_id_follower: item.user_id_follower
+       }))
+     ];
+   }
+   return [{separator: true}, ...newMapUnfollowedDomain];
+ };
+
+ const getFirstData = () => {
+   if (isFirstTimeOpen) {
+     return followingDomains.map((item) => ({
+       ...item,
+       user_id_follower: item.user_id_follower
+     }));
+   }
+   return newMapFollowedDomain;
+ };

+ const data = getData();
+ const firstData = getFirstData();
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->